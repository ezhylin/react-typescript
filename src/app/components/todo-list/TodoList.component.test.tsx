import React from 'react';
import { render } from '@testing-library/react';
import { TodoList, TodoListProps } from './TodoList.component';
import { TodoModel } from './todo.model';

describe('TodoList component', () => {
  const todos: TodoModel[] = [{ id: 0, title: 'title', description: 'descriptoin' }];
  const createTodoList = (extraProps?: TodoListProps) => <TodoList items={todos} {...extraProps} />;

  it('should renders', () => {
    const { asFragment } = render(createTodoList());

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render list of todos', () => {
    const { getAllByTestId } = render(createTodoList());
    const todos = getAllByTestId('todo-item');

    expect(todos.length).toBe(1);
  });
});

export {};
