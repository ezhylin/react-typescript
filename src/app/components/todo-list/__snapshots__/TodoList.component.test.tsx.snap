// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component should renders 1`] = `
<DocumentFragment>
  <ul
    class="TodoList"
  >
    <li
      class="TodoItem"
      data-testid="todo-item"
    >
      <h3>
        title
      </h3>
      <p>
        descriptoin
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
